rule-engine/rule-engine-components/src/main/java/org/thingsboard/rule/engine/metadata/CalculateDeltaNode.java

```diff
@@ -96,6 +99,11 @@ public void onMsg(TbContext ctx, TbMsg msg) {
 
                     BigDecimal delta = BigDecimal.valueOf(previousData != null ? currentValue - previousData.value : 0.0);
 
+                    if (config.isOnlyComputeTrueDeltas() && delta.doubleValue() == 0) {
```

Comment:
if tellFailureIfDeltaIsNegative is set to false while computeOnlyTrueDeltas will be set to true and the delta value is negative then the result will be invalid, we will see a -1 delta value in the output. True deltas means positive value.